src/test/kotlin/racingcar/domain/CarsTest.kt

```diff
@@ -2,57 +2,55 @@ package racingcar.domain
 
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.jupiter.api.Test
-import org.junit.jupiter.api.assertDoesNotThrow
-import org.junit.jupiter.api.assertThrows
+import racingcar.domain.movestrategy.MoveStrategy
 
 internal class CarsTest {
 
-    @Test
-    fun `elements numbers 사이즈가 다를 경우`() {
-        val car1 = Car(Name("test"))
-        val car2 = Car(Name("test"))
-        assertThrows<IllegalArgumentException> { Cars(listOf(car1, car2)).move(listOf()) }
+    private val moveStrategy = object : MoveStrategy {
+        override fun movable() = true
     }
 
-    @Test
-    fun `elements numbers 사이즈가 같을 경우`() {
-        val car1 = Car(Name("test"))
-        val car2 = Car(Name("test"))
-        assertDoesNotThrow { Cars(listOf(car1, car2)).move(listOf(Number(), Number())) }
+    private val notMoveStrategy = object : MoveStrategy {
+        override fun movable() = false
     }
 
     @Test
     fun `움직이지 않는 경우`() {
         val car1 = Car(Name("test"))
         val car2 = Car(Name("test"))
         val cars = Cars(listOf(car1, car2))
-        val movedCars = cars.move(listOf(Number(3), Number(3)))
-        assertThat(cars).isEqualTo(movedCars)
+        val notMovedCars = cars.move(notMoveStrategy)
+
+        assertThat(cars).isEqualTo(notMovedCars)
     }
 
     @Test
     fun `움직이는 경우`() {
         val car1 = Car(Name("test"))
         val car2 = Car(Name("test"))
         val cars = Cars(listOf(car1, car2))
-        val movedCars = cars.move(listOf(Number(4), Number(4)))
+        val movedCars = cars.move(moveStrategy)
+
         assertThat(cars).isNotEqualTo(movedCars)
+        val movedCarDistance = movedCars.elements
+            .map { it.distance }
+        assertThat(movedCarDistance).contains(Distance(1))
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        assertThat(movedCars.elements).allMatch { it.distance == Distance(1) }
```